lc/src/lib.rs

```diff
@@ -1,14 +1,7 @@
-#![no_main]
-sp1_zkvm::entrypoint!(main);
-
 use bitcoin::BtcBlocks;
-use sp1_zkvm::io::{read, write};
-use std::str::FromStr;
-
-fn main() {
-    let blocks = BtcBlocks::from_str(&read::<String>()).expect("to unwrap btc blocks");
 
-    let result = !blocks.iter().enumerate().fold(false, |acc, (i, block)| {
+pub fn prove_btc_blocks(blocks: BtcBlocks) -> bool {
```

Comment:
For eg in the `bitcointee` repo for Mauro yesterday I just used:

```rust
    if !result {
        Err(BtcError::ValidationError)
    } else {
        let last_block_hash = blocks.last().ok_or_else(|| BtcError::NoLastBlock)?.header.block_hash();
        Ok(last_block_hash.to_string())
    }
```

Reply:
In a possibly less elegant way than what you're suggesting, I attempted to add output here:
https://github.com/pnetwork-association/bitcoinz/tree/add-output-to-lc
I wondered whether to return a `Result` right away, eventually I resolved to change to
```rust
pub fn prove_btc_blocks(blocks: BtcBlocks) -> Proof {
```
where
```rust
pub enum Proof {
    False,
    True(BlockHash, BlockHash),
}
```
given that none of the returns inside the function seem fallible at the moment. At a later time, when introducing fallible returns, or either way because it does make sense, that's a sensible direction to take, agreed!